### PR TITLE
Replace oslo.log by daiquiri

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Comment by chungg** [comment]
just curious, what diff between this is vs oslo.log? tbh, we're 3 people; if we keep creating our own stuff, it's going to be pain in the ass to extend or fix ourselves.

**Comment by jd** [comment]
Completely agree. I checked on PyPI and did not find anything that'd do what we needed in oslo.log.

oslo.log is poorly managed and OpenStack specific. For example it logs by default things related to oslo.context or "instances" when, as you might know, Gnocchi is not nova. And there's no way to change that (I could find the thread 1-2 years ago where I tried).

So I extracted what was good in oslo.log and made daiquiri, which is a 250 SLOC setup library around Python logging builtin module.

The alternative is to ditch oslo.log and manually setup the Python logging module, but… since it's just what daiquiri does, it seemed better to build a library so I can use it in other places (e.g. pifpaf, git-pull-request, etc). :)

**Comment by jd** [comment]
@sileht Could you review this and daiquiri and let me know what you think of it?

**Comment by jd** [comment]
Pull-request updated, HEAD is now 7aa8fcef7776de04dc15f4e9a2a2a52f22a794c0

File: gnocchi/service.py [hunk 1]
```diff
@@ -85,9 +102,6 @@ def prepare_service(args=None, conf=None,
                                                 'rest', 'policy.json'))
     conf.set_default('policy_file', cfg_path, group='oslo_policy')
 
-    log.set_defaults(default_log_levels=log.get_default_log_levels() +
-                     ["passlib.utils.compat=INFO"])
```
Comment (chungg) [inline gnocchi/service.py]: just curious, are we still able to control the logging level of our dependencies? ie we only want ERROR from dependencies?

File: gnocchi/opts.py [hunk 2]
```diff
@@ -59,6 +59,49 @@ def __getitem__(self, key):
...
+                'debug',
+                short='d',
+                default=False,
+                help='If set to true, the logging level will be set to '
+                'DEBUG.'),
+            cfg.BoolOpt(
+                'verbose',
+                short='v',
+                default=False,
+                deprecated_for_removal=True,
```
Comment (chungg) [inline gnocchi/opts.py]: what's this deprecated option for?

File: gnocchi/service.py (hunk 1 above)
Comment (jd) [inline gnocchi/service.py]: Yes we can. There's a function to help with that in daiquiri, though it does not seem we really want it. The logging is set by default to WARNING for everything – which ought to be enough, and INFO/WARNING/DEBUG for Gnocchi depending on the options.

File: gnocchi/opts.py (hunk 2 above)
Comment (jd) [inline gnocchi/opts.py]: Hm it's deprecated in oslo.log so I think it's why I pasted it. But the option is not actually used, I need to fix that.

**Comment by jd** [comment]
Pull-request updated, HEAD is now f1a60a56fa599bbf2e89a056ca4e03a3d1f34184

File: gnocchi/opts.py [hunk 3]
```diff
@@ -59,6 +59,48 @@ def __getitem__(self, key):
...
+                'log-file',
+                metavar='PATH',
+                help='(Optional) Name of log file to send logging output to. '
+                'If no default is set, logging will go to stderr as '
+                'defined by use_stderr.'),
+            cfg.BoolOpt(
+                'use-syslog',
+                default=False,
+                help='Use syslog for logging. '
+                'Existing syslog format is DEPRECATED '
```
Comment (chungg) [inline gnocchi/opts.py]: is this irrelevant too?

Comment (jd) [inline gnocchi/opts.py]: It's used in service.py to setup sylog output.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 00327116fe05cae9edc24dbf4ceeef19ef7ff7b7

File: gnocchi/opts.py (hunk 3 above)
Comment (chungg) [inline gnocchi/opts.py]: the deprecated message i mean?

Comment (jd) [inline gnocchi/opts.py]: Oh sorry. Removing.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 7f9517db6af91acd9b2cec62509025e3a0038b13

**Comment by jd** [comment]
Pull-request updated, HEAD is now 8cf60ec915aa6e3fbf3fbce1a6ab7be13a41a35a

File: releasenotes/notes/oslo.log-removal-69a17397b10bc2bb.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    The logging library oslo.log has been removed of daiquiri. Some superflous
```
Comment (chungg) [inline releasenotes/notes/oslo.log-removal-69a17397b10bc2bb.yaml]: s/of daiquiri/for daiquiri/

s/superflous/superfluous/

Comment (jd) [inline releasenotes/notes/oslo.log-removal-69a17397b10bc2bb.yaml]: Fixed, thanks!

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1b829abeb35f98c84fbc6d82383f277130d3b7d0

**Comment by jd** [comment]
Pull-request updated, HEAD is now cad787adc07c35a1a11b4f0a3826b324abd89fab

**Comment by jd** [comment]
Pull-request updated, HEAD is now 58a7437829b6313f98ae2df935d737a1c2447f47

**Comment by jd** [comment]
Pull-request updated, HEAD is now c08e72c7679f536697a167c76be2071601482a1b

**Comment by jd** [comment]
Pull-request updated, HEAD is now f3408ddc44c62e24fcb2b3cc88afae9a5bd3eb7e

**Comment by jd** [comment]
Pull-request updated, HEAD is now 33d7f2c10641e021493e4bebdaa60ba23f4d8da9

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1f1b5fbdede1a2b0ca6364df67f7a84dab3c71be

**Comment by jd** [comment]
Pull-request updated, HEAD is now 61751f20ef051e7bf083076795d5b645a91abdaa

**Comment by jd** [comment]
Pull-request updated, HEAD is now 62d3b61baecde421327ef9ce598843dcc0cdaf0f

**Comment by jd** [comment]
Pull-request updated, HEAD is now b516065b7121ccd5ee77906e45487980fbb5e78f

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6b9b75d32586dd1b289131d6275436e2a5231664

**Comment by jd** [comment]
Pull-request updated, HEAD is now 6276144777018dbb04c0e7d48593976d7693f77d

**Comment by jd** [comment]
Pull-request updated, HEAD is now ad2484d61cab955cdd6f8ae5740ea4049b710ba0

**Pull request merged**